tools/ios/build_xcframework.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+
+# build an xcframework from individual per-platform/arch static frameworks
+
+import argparse
+import pathlib
+import shutil
+import subprocess
+import typing
+
+script_dir = pathlib.Path(__file__).resolve()
+repo_dir = script_dir.parents[2]
+
+default_platform_archs = {
+    "iphoneos": ["arm64"],
+    "iphonesimulator": ["x86_64", "arm64"],
+}
+
+cmake = "cmake"
+lipo = "lipo"
+xcrun = "xcrun"
+
+
+def _get_opencv_toolchain_file(platform: str, opencv_dir: pathlib.Path):
+    return (
+        opencv_dir
+        / "platforms/ios/cmake/Toolchains"
+        / ("Toolchain-iPhoneOS_Xcode.cmake" if platform == "iphoneos" else "Toolchain-iPhoneSimulator_Xcode.cmake")
+    )
+
+
+def _run(cmd_args: typing.List[str], **kwargs):
+    import shlex
+
+    print(f"Running command:\n  {shlex.join(cmd_args)}")
+    subprocess.run(cmd_args, check=True, **kwargs)
+
+
+def _rmtree_if_existing(dir: pathlib.Path):
+    try:
+        shutil.rmtree(dir)
+    except FileNotFoundError:
+        pass
+
+
+def build_framework_for_platform_and_arch(
+    build_dir: pathlib.Path, platform: str, arch: str, config: str, opencv_dir: pathlib.Path, ios_deployment_target: str
+) -> pathlib.Path:
+    build_dir.mkdir(parents=True, exist_ok=True)
+
+    # generate build files
+    generate_args = [
+        cmake,
+        "-G=Xcode",
+        f"-S={repo_dir}",
+        f"-B={build_dir}",
+        "-DCMAKE_SYSTEM_NAME=iOS",
+        f"-DCMAKE_OSX_DEPLOYMENT_TARGET={ios_deployment_target}",
+        f"-DCMAKE_OSX_SYSROOT={platform}",
+        f"-DCMAKE_OSX_ARCHITECTURES={arch}",
+        f"-DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO",
+        "-DOCOS_BUILD_APPLE_FRAMEWORK=ON",
+        # our version of sentencepiece doesn't support iOS build
+        #   CMake Error at out/ios/RelWithDebInfo/_deps/spm-src/src/CMakeLists.txt:288 (install):
+        #     install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
+        #     target "spm_encode".
+        "-DOCOS_ENABLE_SPM_TOKENIZER=OFF",
+        # OpenCV CMake config
+        # tell OpenCV to build zlib so we can link to the static library
+        "-DBUILD_ZLIB=ON",
```

Comment:
should we turn on this flag in externals/opencv.cmake?

Reply:
not sure if this is something we generally want to enable. it's just for the iOS static framework build now.

Reply:
I mean something like 
if (iOS)
set(BUILD_ZLIB ON CACHE INTERNAL "")
endif()
and CPU_BASELINE and IOS_ARCH problem should be same.

Right now, none of internal flag was exposed for build-script and all of them are managed by CMakefile. it would be better to keep the same pattern.

Reply:
moved some of them into opencv.cmake. but others are required by the toolchain file so setting it in opencv.cmake would be too late.